### PR TITLE
Codacy: fix further issues

### PR DESCRIPTION
**Pull request by AndyScherzinger**
* fixing further issues reported by codacy: https://www.codacy.com/app/Nextcloud/android/issues
* move helper functions to a separate utils class and clean up dependencies between the classes
* cleanup upload list adapter

local tests on my phone where successful 😺

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@mario @tobiasKaminsky please review :)

**Comment by AndyScherzinger** [comment]
pinging @tobiasKaminsky @mario for review :)

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger can you rebase it and then we can review it :-)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky Done :)

File: src/main/java/com/owncloud/android/authentication/AccountUtils.java [hunk 1]
```diff
@@ -202,45 +200,137 @@ public static void resetOwnCloudAccount(Context context) {
...
+     * @param context Used to access the AccountManager.
      */
-    public static String getWebdavPath(OwnCloudVersion version, String authTokenType) {
-        if (version != null) {
-            if (AccountTypeUtils.getAuthTokenTypeAccessToken(MainApp.getAccountType()).equals(authTokenType)) {
-                return ODAV_PATH;
-            }
-            if (AccountTypeUtils.getAuthTokenTypeSamlSessionCookie(MainApp.getAccountType()).equals(authTokenType)) {
-                return SAML_SSO_PATH;
+    public static void updateAccountVersion(Context context) {
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/authentication/AccountUtils.java]: Don't we have eliminated this a while ago?

Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/authentication/AccountUtils.java]: Don't know but checked and it isn't used anywhere (anymore) so I removed the code + the used private method via https://github.com/nextcloud/android/pull/1357/commits/0e129dd8392d49deb429770a8f858ef4a64bf816

File: src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java [hunk 2]
```diff
@@ -151,7 +150,7 @@ public void onCreate(Bundle savedInstanceState) {
         mFileListFragment = (LocalFileListFragment) getSupportFragmentManager().findFragmentById(R.id.local_files_list);
         
         // Set input controllers
-        mCancelBtn = (Button) findViewById(R.id.upload_files_btn_cancel);
+        Button mCancelBtn = (Button) findViewById(R.id.upload_files_btn_cancel);
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java]: Theoretically this is then just a regular object so no "m" as this indicates members variables of the class ;-)

File: src/main/java/com/owncloud/android/ui/adapter/ProgressListener.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Nextcloud Android client application
+ *
+ * @author Andy Scherzinger
+ * Copyright (C) 2016 Andy Scherzinger
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/adapter/ProgressListener.java]: 2016?

File: src/main/java/com/owncloud/android/ui/fragment/SearchShareesFragment.java [hunk 4]
```diff
@@ -183,7 +182,7 @@ public void refreshUsersOrGroupsListFromDB (){
     private void updateListOfUserGroups() {
         // Update list of users/groups
         // TODO Refactoring: create a new {@link ShareUserListAdapter} instance with every call should not be needed
-        mUserGroupsAdapter = new ShareUserListAdapter(
+        ShareUserListAdapter mUserGroupsAdapter = new ShareUserListAdapter(
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/SearchShareesFragment.java]: Same here.

**Comment by tobiasKaminsky** [comment]
also a very minor thing: there are some lines that are > 120.

**Comment by tobiasKaminsky** [comment]
👍

File: src/main/java/com/owncloud/android/ui/adapter/ProgressListener.java (hunk 3 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/adapter/ProgressListener.java]: Not newly created, it is "just" an inner class moved into its own file :)

File: src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java (hunk 2 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java]: Fixed via https://github.com/nextcloud/android/pull/1357/commits/41b685043810ca031eb2d158fb99db8ee8778207

**Comment by AndyScherzinger** [comment]
Fixed your comments and also shortened the lines > 120 character I found.

**Comment by AndyScherzinger** [comment]
So pinging @mario for review :) Thanks in advance ❤️

**Comment by mario** [comment]
👍

**Pull request merged**